pkg/aetosutils/dashboardutils.go

```diff
@@ -0,0 +1,554 @@
+	}
+}
+
+// TestCaseEnd update testcase  to dashboard DB
+func (d *Dashboard) TestCaseEnd() {
+	if d.IsEnabled {
+		var url string
+		if d.testcaseID == 0 {
+			d.TestLog.Error("TestCaseID is empty")
+			return
+		}
+		if FailureCause != "" {
+			url = fmt.Sprintf("%s/testcase/%d/end?cause=%s", DashBoardBaseURL, d.testcaseID, FailureCause)
+		} else {
+			url = fmt.Sprintf("%s/testcase/%d/end", DashBoardBaseURL, d.testcaseID)
+		}
+		resp, respStatusCode, err := rest.PUT(url, nil, nil, nil)
+		if err != nil {
+			d.TestLog.Errorf("Error in updating TestCase, Caose: %v", err)
+		} else if respStatusCode != http.StatusOK {
+			d.TestLog.Errorf("Failed to end TestCase, Resp : %s", string(resp))
+		} else {
+			d.TestLog.Infof("TestCase %d ended successfully", d.testcaseID)
+		}
+
+		var updateResponse testCaseUpdateResponse
+		err = json.Unmarshal(resp, &updateResponse)
+		if err != nil {
+			d.TestLog.Errorf("Error parsing update test output, %v", err)
+		}
+		result := updateResponse.TestCaseStatus
+		d.VerifySafely(result, "PASS", "Workflow completed successfully ?")
+		d.TestLog.Info("--------Test End------")
+		d.TestLog.Infof("#Test: %s ", testCase.ShortName)
+		d.TestLog.Infof("#Description: %s ", testCase.Description)
+		d.TestLog.Infof("#Result: %s ", result)
+		d.TestLog.Info("------------------------")
+		removeTestCaseFromStack(d.testcaseID)
+		if len(testCasesStack) > 0 {
+			d.testcaseID = testCasesStack[len(testCasesStack)-1]
+		}
+	}
+}
+
+func removeTestCaseFromStack(testcaseID int) {
+
+	removeIndex := -1
+	for i, v := range testCasesStack {
+		if v == testcaseID {
+			removeIndex = i
+			break
+		}
+	}
+
+	if removeIndex != -1 {
+		testCasesStack = append(testCasesStack[:removeIndex], testCasesStack[removeIndex+1:]...)
+	}
+
+}
+
+// TestSetUpdate update test set  to dashboard DB
+func (d *Dashboard) TestSetUpdate(testSet *TestSet) {
+
+	if d.IsEnabled {
+
+		if d.TestSetID == 0 {
+			d.TestLog.Error("TestSetID is empty")
+		}
+
+		updateTestSetURL := fmt.Sprintf("%s/testset/%d", DashBoardBaseURL, d.TestSetID)
+		resp, respStatusCode, err := rest.PUT(updateTestSetURL, testSet, nil, nil)
+
+		if err != nil {
+			d.TestLog.Errorf("Error in updating TestSet, Caose: %v", err)
+		} else if respStatusCode != http.StatusOK {
+			d.TestLog.Errorf("Failed to update TestSet, Resp : %s", string(resp))
+		} else {
+			d.TestLog.Infof("TestSetId %d update successfully", d.TestSetID)
+
+		}
+	}
+}
+
+// TestCaseBegin start the test case and push data to dashboard DB
+func (d *Dashboard) TestCaseBegin(testName, description, testRepoID string, tags map[string]string) {
+	if d.IsEnabled {
+		if d.TestSetID == 0 {
+			d.TestLog.Errorf("In testset begin, TestSetID is empty, cannot update update testcase")
+			return
+		}
+
+		testCase = TestCase{}
+		_, file, _, ok := runtime.Caller(1)
+		if ok {
+			m := regexp.MustCompile(`stork`)
```

Comment:
Which files are being filtered using this regex ?

Reply:
anything under /stork/* folder. Thats how the testcase name gets `TestHealthMonitor/stopDriverTest` name.